### ISSUE TITLE
minDate and maxDate not adhering to UTC offset when select is missing

### ISSUE THREAD

**Issue by eastwood** [Hacker0x01/react-datepicker#747]
After some preliminary investigation, it seems that when there is no `select` prop present, the `minDate` and `maxDate` values to not adhere to the `utcOffset`.

Demo project: https://github.com/eastwood/react-datepicker-fix.
I'd expect to have the same `disabled` days for the following:

1. Australia/Sydney time is 24th February 00:00 UTC+10
2. America/Los Angeles time is 23rd February 05:00 UTC-8
3. If I'm in UTC-8 (Los Angeles) time and the following settings are set
2. Set `minDate={moment()}`
3. Set `selected={null}`
4. Set `utcOffset={10}`

I'd expect to only be able to choose dates from the 24th February onwards.
Unfortunately, the 23rd February is enabled.
However, if I set `selected` to anything, i.e `selected={moment().subtract(2, 'days')}`, then the correct dates will be disabled.

To test this:

1. Clone repo
2. Change system timezone accordingly
3. Start application with `npm run start`
4. Note that the I have the `utcOffset=10`. So feel free to play with it to cross a day boundary.

I'll do some more investigation into this and happy to submit PR afterwards.

**Comment by martijnrusschen** [comment]
+1 on more investigation and a fix :) Happy to help reviewing any solutions for you.

**Comment by dvdgarcia** [comment]
+1 The issue as well happens without set the offset.

For example if in excludeDates a I exclude tomorrow and next tomorrow when I open the datepicker I see today and tomorrow disabled, then I select next tomorrow, the date picker is closed I open again and I see what I was expecting originally(tomorrow and next tomorrow disabled)

On my case I have the version 0.39.0

<DatePicker dateFormat="DD/MM/YY" minDate={moment()} className="form-control" selected={this.props.value ? moment.unix(this.props.value) : null} placeholderText={this.props.placeholder} onChange={::this.handleChange}
excludeDates={this.props.excludeDates?this.props.excludeDates:[]}/>;

**Comment by aij** [comment]
@dvdgarcia What time zone are you using for excludeDates?

Things get a little screwy when mixing moments in different UTC offsets. (Which is especially easy to do accidentally when you are in a time zone with daylight savings time...)

**Comment by dvdgarcia** [comment]
I am in Spain so Madrid+1.

I have to say that I achieved a way for working, if selected value is not null works fine. So the bug is this one:
with this implementation with excludedDates in the same UTC offset:
```
<DatePicker dateFormat="DD/MM/YY"
 minDate={moment()} 
className="form-control" 
selected={this.props.value ? moment.unix(this.props.value) : null} 
placeholderText={this.props.placeholder} 
onChange={::this.handleChange}
excludeDates={this.props.excludeDates?this.props.excludeDates:[]}/>
```
I open the date picker and the dates excluded are wrong, I select one, the datepicker is closed, then I open it again and I see the disabled dates as I wanted originally.

So I tried to do this implementation:

<DatePicker dateFormat="DD/MM/YY"
 minDate={moment()} 
className="form-control" 
selected={this.props.value ? moment.unix(this.props.value) : moment()}  <---THIS IS THE CHANGE 
placeholderText={this.props.placeholder} 
onChange={::this.handleChange}
excludeDates={this.props.excludeDates?this.props.excludeDates:[]}/>


I initialize the selected value with moment() and it works perfect.
